pkg/distributor/distributor.go

```diff
@@ -376,7 +386,8 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 			n := 0
 			pushSize := 0
 			prevTs := stream.Entries[0].Timestamp
-			addLogLevel := validationContext.allowStructuredMetadata && validationContext.discoverLogLevels && !lbs.Has(labelLevel)
+
+			addLogLevel := validationContext.allowStructuredMetadata && validationContext.discoverLogLevels && !hasAnyLevelLabels(lbs)
```

Comment:
Since this is for standardisation I think that if the label exist and is found it should be still added a structured metadata under `detected_level` this way we can have a query that works for everyone.

Reply:
added this change